packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1659,16 +1679,40 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
-    public async createDataStore(pkg: string | string[]): Promise<IFluidRouter> {
-        return this._createDataStore(pkg, false /* isRoot */);
+    public async createDataStore(pkg: string | string[]): Promise<IDataStore> {
+        const internalId = uuid();
+        return channelToDataStore(
+            await this._createDataStore(pkg, false /* isRoot */, internalId),
+            internalId,
+            this,
+            this.mc.logger);
     }
 
-    public async createRootDataStore(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
+    private async createRootDataStoreLegacy(pkg: string | string[], rootDataStoreId: string): Promise<IFluidRouter> {
```

Comment:
The name is a bit misleading as it's used for detached flow as well. Maybe fine to leave it as is, but comments about each function would be nice :)

Reply:
Agree with the comments. And they're incoming. But I'm confused about this instance. This is a private method only called by `createRootDataStore` when the feature gate is off.